### PR TITLE
Replace FileSystem Ledger with Rocks DB

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radix-engine/src/engine/track.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ pub struct Track<'l, L: Ledger> {
     code_cache: LruCache<Address, Module>, // TODO: move to ledger level
 }
 
-impl<'l, L: Ledger> Track<'l, L> {
+impl<'l, L: SubstateStore> Track<'l, L> {
     pub fn new(
         ledger: &'l mut L,
```
Comment (gianlucardx) [inline radix-engine/src/engine/track.rs:45]: Since we're renaming Ledger to SubstateStore, should the various L and ledger reference be removed?

**Review by gianlucardx — COMMENTED** [review]
very minor comments

File: radix-engine/src/ledger/memory.rs [hunk 2]
```diff
@@ -40,7 +40,7 @@ impl Default for InMemoryLedger {
     }
 }
 
-impl Ledger for InMemoryLedger {
+impl SubstateStore for InMemoryLedger {
```
Comment (iamyulong) [inline radix-engine/src/ledger/memory.rs:43]: I'd suggest we rename `InMemoryLedger` too.

File: simulator/src/ledger/radix_engine_db.rs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+use radix_engine::model::*;
+use scrypto::types::*;
+
+pub struct RadixEngineDB {
+    db: DBWithThreadMode<SingleThreaded>,
+}
+
+impl RadixEngineDB {
+    pub fn new(root: PathBuf) -> Self {
+        let db = DB::open_default(root.as_path()).unwrap();
```
Comment (iamyulong) [inline simulator/src/ledger/radix_engine_db.rs:14]: I wonder what happens if the folder already exists but not a database. Most developers will likely have this setup. Need to advise them how to migrate.

Comment (talekhinezh) [reply simulator/src/ledger/radix_engine_db.rs:14]: Default settings creates a new database if empty

**Review by iamyulong — APPROVED** [review]

**Pull request merged**